src/sign/utils.ts

```diff
@@ -9,12 +9,19 @@ export function getHash(data) {
 
 export async function verify(address, sig, data) {
   const { domain, types, message } = data;
-  const recoverAddress = verifyTypedData(domain, types, message, sig);
+
   const hash = getHash(data);
   console.log('Hash', hash);
   console.log('Address', address);
-  console.log('Recover address', recoverAddress);
-  if (address === recoverAddress) return true;
+
+  try {
+    const recoverAddress = verifyTypedData(domain, types, message, sig);
+    console.log('Recover address', recoverAddress);
+    if (address === recoverAddress) return true;
+  } catch (e) {
+    console.log('Could not recoverAddress:' + e.message);
+  }
+
```

Comment:
It can catch all other unknown errors too?

Reply:
Well, there is a lot going on in verifyTypedData, and I would classify it in 2 categories

1 : (This is the error we should catch) the signature validation category, checking things like 
  - ecrecover
  - signature length
  - rsv matching
   
2 : the 712 data validation like checking if the domain / types / message are 712 compliant

what matters here is to catch the 1st category as ecrecover, siglen and rsv matching would fail in the case of 1271

it also however catches the typed data compliance error, which you might want to throw.


As I wanted to keep the PR as small as possible, I kept your original call to verifyTypedData, wrapping it in a try catch but if you do care about throwing a relevant error related to the 712 compliant data and separate the 2 types of errors,

I would suggest something like that:

```js
// first, checking the data structure (which is independent from the sig)
try {
TypedDataEncoder.hash(domain , types, message)
} catch (e) {
 throw Error('Invalid 712 data structure')
}

// second, continue with the original process, matching the message AND the signature
try {
  const recoverAddress = verifyTypedData(domain, types, message, sig);
  console.log('Recover address', recoverAddress);
  if (address === recoverAddress) return true;
} catch (e) {
  console.log('Could not recoverAddress:' + e.message);
}
```